doc/SAI-Proposal-Packet-Trimming.md

```diff
@@ -0,0 +1,142 @@
+                                                                                                    │ │ │ │ │        │                  
+                                                                                   ┌────────────────► │ │ │ │        │                  
+                                                                                   │                │ │ │ │ │        │    Queue         
+                                                                                   │                │ │ │ │ │        │                  
+                                                                                   │                │ │ │ │ │        │                  
+                                                                                   │                └─┴─┴─┴─┴────────┘                  
+     ┌──────────────┐                                                              │                                                    
+     │              │  ┌──────────────────────────────────────────────────────┐    │                ┌─┬─┬─┬─┬─┬─┬─┬─┬┐                  
+     │              │  │                                                      │    │                │ │ │ │ │ │ │ │ ││                  
+     │              │  │                                                      │    │     \   /      │ │ │ │ │ │ │ │ ││                  
+     │              │  │                                                      │    │      \ /       │ │ │ │ │ │ │ │ ││                  
+     │    Packet    │  │           Pipeilne                                   ┼────┼───────\────────► │ │ │ │ │ │ │ ││     Queue        
+     │              │  │                                                      │           / \       │ │ │ │ │ │ │ │ ││                  
+     │              │  │                                                      │          /   \      │ │ │ │ │ │ │ │ ││                  
+     │              │  └──────────────────────────────────────────────────────┘                     └─┴─┴─┴─┴─┴─┴─┴─┴┘                  
+     │              │                                                                                                                   
+     │              │                                                                                                                   
+     │              │                                                                                                                   
+     └──────────────┘                                                                                                                   
+```
+
+This feature assumes that forwarding tables are configured properly, and the original packet would be delivered to the destination successfully if not for the congestion.
+
+## Spec
+There is a tradeoff between trying to configure a higher threshold in a queue buffer profile and trimming the packet.
```

Comment:
A static trimming threshold may not be effective with shared buffer switches, where the buffer resources allocated to a queue or port can vary over time. Therefore, we propose adding a new attribute to a buffer profile to allow configuring packet trimming on such stricter profiles: